### PR TITLE
Dev-docs: Getting Started

### PR DESCRIPTION
**Pull request by mdoucet**
Move over the following:

  * http://developer.mantidproject.org/GettingStarted.html. 
   *  [Checking out the Mantid Code](http://www.mantidproject.org/Checking%20out%20the%20Mantid%20Code)
   *  [Building the Mantid Framework Code](http://www.mantidproject.org/Building%20the%20Mantid%20Framework%20Code)
   *  [Building with CMake](http://www.mantidproject.org/Building%20with%20CMake)
   *  [Building VATES](https://www.mantidproject.org/Building_VATES)
   *  [Building on osx]([https://github.com/mantidproject/mantid/wiki/Building-Mantid-on-OS-X-10.9-&-10.10-using-clang-and-Xcode)

**To test:**

- Read the docs.

Refs #22082

**Release Notes** 
*Does not need to be in the release notes.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: dev-docs/source/CheckingOutTheMantidCode.rst [hunk 1]
```diff
@@ -0,0 +1,117 @@
+.. _CheckingOutTheMantidCode:
```
Comment (DanNixon) [inline dev-docs/source/CheckingOutTheMantidCode.rst]: Should this file be here? This information is duplicated in `GettingStarted.rst`.

File: dev-docs/source/GettingStarted.rst [hunk 2]
```diff
@@ -56,3 +54,104 @@ package and install it:
...
+The build environment on OS X is described here :ref:`BuildingOnOSX`.
+
+Checking out the Mantid code
+############################
+We use `Git`_ as our version control system (VCS). The master copies of our repositories are located at `GitHub <http://github.com/mantidproject>`_. We have a number of repositories, of which the main one (the one containing all the source code for Mantid itself) is called simply `mantid <http://github.com/mantidproject/mantid>`_.
+
+Naturally, you will require git installed on your machine. Windows and Mac OSX downloads are available at https://www.git-scm.com/; Linux users should install via their package manager. This will give you access to git via the command line, with the Windows and Mac installers also including a simple graphical tool (Git Gui). Other graphical tools include `TortoiseGit <http://code.google.com/p/tortoisegit/>`_ (for windows) and a Visual Studio extension. There are others (do add your favourite!). Git is also available within Eclipse and QtCreator.
+
+Set up Git
+----------
```
Comment (DanNixon) [inline dev-docs/source/GettingStarted.rst]: This should just point to external documentation, e.g. https://git-scm.com/book/en/v2/Getting-Started-First-Time-Git-Setup

File: dev-docs/source/GettingStarted.rst [hunk 3]
```diff
@@ -56,3 +54,104 @@ package and install it:
...
+            prompt = false
+    [difftool "sourcetree"]
+            cmd = opendiff \"$LOCAL\" \"$REMOTE\"
+            path = 
+    [mergetool "sourcetree"]
+            cmd = /Applications/SourceTree.app/Contents/Resources/opendiff-w.sh \"$LOCAL\" \"$REMOTE\" -ancestor \"$BASE\" -merge \"$MERGED\"
+            trustExitCode = true
+
+Extra Windows Step (Do Not Skip This!)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (DanNixon) [inline dev-docs/source/GettingStarted.rst]: This has already been mentioned above.

File: dev-docs/source/BuildingWithCMake.rst [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+* It should be rare that you will need to edit the CMake build (``CMakeLists.txt``) files. The most common occurrence will be when you add a new file. This must be added to the corresponding CMakeLists file, e.g. if you add a file to Kernel, edit ``Mantid/Framework/Kernel/CMakeLists.txt`` to add the source, header and test files to the long lists of filepaths at the top of the file.
+* The class maker utility (:ref:`UsefulTools`) can edit the ``CMakeList.txt`` for you automatically
+* There are similar places in the Qt projects for ui files and files that need moc-ing.
+* If you add a new dependency, that will need to be added (this is less straightforward - do ask for help).
+* Cache variables can be added via the CMake Gui or by running ``ccmake``.
+
+Building the installer package
+##############################
+
+* For WIndows only, you first need to install NSIS, available at: http://nsis.sourceforge.net/Download. Ensure that the install directory is added to the PATH. You should be able to type ``makensis /?`` in a command prompt.
```
Comment (DanNixon) [inline dev-docs/source/BuildingWithCMake.rst]: `WIndows` -> `Windows`

File: dev-docs/source/BuildingWithCMake.rst [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+As RHEL6 contains versions of various software that are out of date, we need to use newer versions.  In order for these new versions to not interfere with the default versions on the system, we install them using something called `Software Collections <https://access.redhat.com/site/documentation//en-US/Red_Hat_Developer_Toolset/1/html/Software_Collections_Guide/index.html>`_.  The basic upshot of using these is that you have to prefix your configure and make commands with ``scl enable mantidlibs34``.  You always need to enclose the command you want to run in single quotes.
+So some examples of commands we might use are:
+
+.. code-block:: sh
+
+  scl enable mantidlibs34 'cmake-gui'
+  scl enable mantidlibs34 'cmake -G"Ninja" /path/to/Mantid'
+  scl enable mantidlibs34 'ninja -j10'
+
+From the command line ...
```
Comment (DanNixon) [inline dev-docs/source/BuildingWithCMake.rst]: The ellipses in headings aren't needed.

File: dev-docs/source/BuildingWithCMake.rst [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+--------------
+
+The default location for the cache directory is ``$HOME/.ccache``. If you're home directory is on a network-mounted drive then the location of this cache be moved to provide the best performance. On newer versions of ``ccache`` run ``ccache --set-config=cache_dir=PATH_TO_CACHE``. Older versions (<3.2) do not allow this and must fall back to setting the ``CCACHE_DIR`` environment variable in your shell profile.
+
+Configuring your build
+######################
+
+CMake encourages the use of 'out of source' builds. This means that all generated files are placed in a separate directory structure to the source files. This separation makes a full clean easier (you just delete everything) and means that you can have different types of build (Release, Debug, different compiler versions, ....) in separate places (N.B. For Visual Studio & XCode, you can still select the type of build from within the IDE).
+
+Red Hat Enterprise Linux (RHEL) 6 Special Notes
```
Comment (DanNixon) [inline dev-docs/source/BuildingWithCMake.rst]: This should probably just be removed since it isn't a supported OS anymore.

File: dev-docs/source/BuildingVATES.rst [hunk 7]
```diff
@@ -0,0 +1,118 @@
+.. _BuildingVATES:
+
+==============
+Building VATES
+==============
+
```
Comment (DanNixon) [inline dev-docs/source/BuildingVATES.rst]: TOC is missing.

File: dev-docs/source/BuildingWithCMake.rst [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+
+Building the installer package
+##############################
+
+* For WIndows only, you first need to install NSIS, available at: http://nsis.sourceforge.net/Download. Ensure that the install directory is added to the PATH. You should be able to type ``makensis /?`` in a command prompt.
+* Run CMake with "ENABLE_CPACK" enabled. If using the GUI you need to click the "Advanced" checkbox to see this option.
+* You will now have a build target called "PACKAGE" available to create the installer package.
+
+Troubleshooting on OSX
+----------------------
```
Comment (DanNixon) [inline dev-docs/source/BuildingWithCMake.rst]: Is this relevant here? There is already a Troubleshooting section on the OS X specific page.

**Review by DanNixon — CHANGES_REQUESTED** [review]
Mostly OK. There is a fair bit of duplicated information.

I feel this would be better if the complete build instructions were merged into a single document that could be followed from start to finish rather than the several separate documents for different stages/platforms.

File: dev-docs/source/CheckingOutTheMantidCode.rst (hunk 1 above)
Comment (mdoucet) [reply dev-docs/source/CheckingOutTheMantidCode.rst]: That's a leftover from someone else, I'll delete that.

File: dev-docs/source/GettingStarted.rst (hunk 2 above)
Comment (mdoucet) [reply dev-docs/source/GettingStarted.rst]: Are you talking about the whole "Set up Git" section?

File: dev-docs/source/GettingStarted.rst (hunk 3 above)
Comment (mdoucet) [reply dev-docs/source/GettingStarted.rst]: This is resolved now that the extra file above has been removed.

File: dev-docs/source/BuildingWithCMake.rst (hunk 8 above)
Comment (mdoucet) [reply dev-docs/source/BuildingWithCMake.rst]: This one is not specific to OSX, and relates to VATES. I can ask our local VATES developer to see if it's relevant enough to keep.

File: dev-docs/source/GettingStarted.rst (hunk 2 above)
Comment (DanNixon) [reply dev-docs/source/GettingStarted.rst]: Yep.

File: dev-docs/source/GettingStarted.rst (hunk 3 above)
Comment (DanNixon) [reply dev-docs/source/GettingStarted.rst]: I mean earlier in this file.

**Comment by mdoucet** [comment]
@DanNixon I mostly agree with what you're saying. I wasn't sure how much I should "rock the boat" in moving over the docs so I opted for leaving the text mostly unchanged.

The problem though is that a single page would be long and hard to follow (just look at all the info on the OSX page). You could simply have three pages (linux/windows/osx) with the understanding that there would be duplication between the three. You could also have one main flow with links to more details info for each OS.

I don't mind either way.

**Comment by martyngigg** [comment]
@DanNixon I agree the current layout is not ideal. Would you be okay if we take it as is so we can get the wiki out of the way and then when the dust settles we can look at a better structure when everything is in one place?

**Review by DanNixon — APPROVED** [review]
